src/tmpl/yarn-project.nix.in

```diff
@@ -46,14 +46,16 @@ let
     # will follow the symlink and consider the location of the original file as
     # the projects root directory.
     cp ${packageJson} package.json
-    ln -s ${yarnLock} yarn.lock
-    ln -s ${yarnrcYaml} .yarnrc.yml
+    if [ ! -f "yarn.lock" ]; then
+      ln -s ${yarnLock} yarn.lock
+    fi
+
+    if [ ! -f ".yarnrc.yml" ]; then
+      ln -s ${yarnrcYaml} .yarnrc.yml
+    fi
```

Comment:
The idea was that these files should not be included in the nix package src. You can restrict what’s included by following this example: https://github.com/rhinofi/launch-deversifi/blob/234ed5257b2e1809feedd74d37ba4d3f61705e80/dvf-stark-trading/package.nix#L12

or by using https://github.com/numtide/nix-filter

Reply:
It’s quite important to provide explicit list of dirs\files to be included to avoid unnecessary rebuilds since you will rarely need [all files from the root](https://github.com/rhinofi/contracts/blob/57fd5bcbae4ca1481f4fb28799a1de8120a3a7db/cross-swap/default.nix#L16) of the project.

Reply:
Removed the conditionals and using `nix-filter` now to get the required files only